### DISCUSSION TITLE
Bug: unable to use keyboard to enter "Enter" key

### DISCUSSION THREAD

**Discussion by IlysvlVEizbr** [AppGround-io/bluetooth-keyboard-and-mouse-support#298] [Bugs & Problems]
Whether it is the "Enter" key in the on-screen soft keyboard or the "Key->Keyboard shortcut" in the app interface, all "Enter" will enter the letter "q" in the text box on the controlled device. It will not change lines or perform confirmation or Goto normally.

**Comment by AppGround-io** [comment]
There appears to be a problem with the configuration of the Bluetooth connection. Try removing the existing Bluetooth pairing on the device you are trying to control and set it up again. If the problem persists, please provide more details about the devices you are using so I can help you further.

**Comment by IlysvlVEizbr** [comment]
Sorry, I found the real cause of the problem, and it's a silly story. I am controlling an Android TV. Whenever I open an input field, the TV system automatically pops up the soft keyboard and gets the focus so that the user can input using the remote control. However, I am actually simulating Bluetooth keyboard input on my phone instead of using the soft keyboard on the TV. As a result, the selected state of the soft keyboard never changes, and by default, the selected letter is "q" in the top left corner of the keyboard. Every time I press "Enter", it is equivalent to pressing the OK button on the remote control, causing the soft keyboard to input a "q" instead of the desired "Enter".

However, is it possible for us to avoid similar situations by optimizing this software and ensuring that we are inputting "Enter"?
